### PR TITLE
chore: add audit to SDK CI

### PR DESCRIPTION
**Pull request by austinabell**
Ignoring time audit vulnerability because it isn't exposed through chrono, which is what `nearcore` crates pull this in through

https://github.com/chronotope/chrono/issues/602

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -50,3 +50,22 @@ jobs:
...
+    name: Audit
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Sources
+        uses: actions/checkout@v2
+      - name: Install Toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: stable
```
Comment (itegulov) [inline .github/workflows/test.yml:63]: Does it matter if we only use stable here vs also MSRV (1.56.0)?

**Review by itegulov — APPROVED** [review]
LGTM. Do we want to add this as a required check for merging PRs?

**Comment by itegulov** [comment]
> Ignoring time audit vulnerability because it isn't exposed through chrono, which is what nearcore crates pull this in through

Also noting that just because `time` is not exposed does not mean that it cannot cause segfaults but yeah makes sense that it should be solved at nearcore side

**Comment by austinabell** [comment]
> > Ignoring time audit vulnerability because it isn't exposed through chrono, which is what nearcore crates pull this in through
> 
> Also noting that just because `time` is not exposed does not mean that it cannot cause segfaults but yeah makes sense that it should be solved at nearcore side

That wasn't what I was saying. I was saying that `chrono` doesn't use the section of code that causes segfaults in `time` and `chrono` is only pulled in through `nearcore` crates. This is why I linked https://github.com/chronotope/chrono/issues/602 to show that the resolution was that the vulnerable APIs are not called.

File: .github/workflows/test.yml (hunk 1 above)
Comment (austinabell) [reply .github/workflows/test.yml:63]: it doesn't affect this since it's just checking the dependencies in the audit registry and not compiling things

**Review by ChaoticTempest — APPROVED** [review]
Also looks good to me. Might be worthwhile to add something similar in workspaces

**Comment by austinabell** [comment]
> Also looks good to me. Might be worthwhile to add something similar in workspaces

It's a very low-cost sanity check. I'd recommend it. I was going to add it sooner but there were previously many vulnerable crates in previous versions of testing utilities we were using through `nearcore`

**Pull request merged**